In [1]:
import torch

In [2]:
torch.cuda.is_available()

True

In [3]:
import os

In [4]:
os.listdir("awd-lstm-lm")

FileNotFoundError: [WinError 3] The system cannot find the path specified: 'awd-lstm-lm'

In [1]:
import model

In [4]:
#with open("PTBC.pt", 'rb') as f:
#with open("e50_PTBC_pretrained.pt", 'rb') as f:
#with open("ENWIK8.pt.e35", 'rb') as f:
    model, criterion, optimizer = torch.load(f)

In [56]:
criterion

SplitCrossEntropyLoss()

In [6]:
model

RNNModel(
  (lockdrop): LockedDropout()
  (idrop): Dropout(p=0.1, inplace=False)
  (hdrop): Dropout(p=0.25, inplace=False)
  (drop): Dropout(p=0.1, inplace=False)
  (encoder): Embedding(50, 200)
  (rnns): ModuleList(
    (0): WeightDrop(
      (module): LSTM(200, 1000)
    )
    (1): WeightDrop(
      (module): LSTM(1000, 1000)
    )
    (2): WeightDrop(
      (module): LSTM(1000, 200)
    )
  )
  (decoder): Linear(in_features=1000, out_features=50, bias=True)
)

In [5]:
model

RNNModel(
  (lockdrop): LockedDropout()
  (idrop): Dropout(p=0.1, inplace=False)
  (hdrop): Dropout(p=0.1, inplace=False)
  (drop): Dropout(p=0.4, inplace=False)
  (encoder): Embedding(89, 400)
  (rnns): ModuleList(
    (0): WeightDrop(
      (module): LSTM(400, 1840)
    )
    (1): WeightDrop(
      (module): LSTM(1840, 1840)
    )
    (2): WeightDrop(
      (module): LSTM(1840, 400)
    )
  )
  (decoder): Linear(in_features=1840, out_features=89, bias=True)
)

In [8]:
model

RNNModel(
  (lockdrop): LockedDropout()
  (idrop): Dropout(p=0.1, inplace=False)
  (hdrop): Dropout(p=0.1, inplace=False)
  (drop): Dropout(p=0.4, inplace=False)
  (encoder): Embedding(89, 400)
  (rnns): ModuleList(
    (0): WeightDrop(
      (module): LSTM(400, 1840)
    )
    (1): WeightDrop(
      (module): LSTM(1840, 1840)
    )
    (2): WeightDrop(
      (module): LSTM(1840, 400)
    )
  )
  (decoder): Linear(in_features=1840, out_features=89, bias=True)
)

In [58]:
model.encoder.__dict__

{'training': False,
 '_parameters': OrderedDict([('weight',
               Parameter containing:
               tensor([[ 0.3571, -0.4133,  0.1617,  ...,  0.3992, -0.5032,  0.3619],
                       [ 0.2212,  0.4594, -0.5342,  ..., -0.3426,  2.2469, -0.0726],
                       [ 0.7304,  0.0063,  0.0855,  ...,  0.2008, -0.5905,  0.1520],
                       ...,
                       [ 0.1620,  0.0122,  0.5074,  ...,  0.2366, -0.3072, -0.3183],
                       [-0.3365, -0.1019,  0.1442,  ..., -0.1671,  0.0168,  0.5306],
                       [-0.0264,  0.0492,  0.1915,  ..., -0.2360,  0.0659, -0.0358]],
                      device='cuda:0', requires_grad=True))]),
 '_buffers': OrderedDict(),
 '_non_persistent_buffers_set': set(),
 '_backward_hooks': OrderedDict(),
 '_is_full_backward_hook': None,
 '_forward_hooks': OrderedDict(),
 '_forward_pre_hooks': OrderedDict(),
 '_state_dict_hooks': OrderedDict(),
 '_load_state_dict_pre_hooks': OrderedDict(),
 '_load_sta

In [59]:
model.decoder.__dict__

{'training': False,
 '_parameters': OrderedDict([('weight',
               Parameter containing:
               tensor([[ 0.3571, -0.4133,  0.1617,  ...,  0.3992, -0.5032,  0.3619],
                       [ 0.2212,  0.4594, -0.5342,  ..., -0.3426,  2.2469, -0.0726],
                       [ 0.7304,  0.0063,  0.0855,  ...,  0.2008, -0.5905,  0.1520],
                       ...,
                       [ 0.1620,  0.0122,  0.5074,  ...,  0.2366, -0.3072, -0.3183],
                       [-0.3365, -0.1019,  0.1442,  ..., -0.1671,  0.0168,  0.5306],
                       [-0.0264,  0.0492,  0.1915,  ..., -0.2360,  0.0659, -0.0358]],
                      device='cuda:0', requires_grad=True)),
              ('bias',
               Parameter containing:
               tensor([ 0.4391,  0.1382,  0.0305,  0.7156, -0.0027,  0.4404, -0.6091,  0.1100,
                        0.5843, -0.1457, -0.0578, -0.7736,  0.4537,  0.0497, -0.4391, -0.2340,
                        0.2636, -0.0671,  0.0217, -0.

In [3]:
loaded_corpus = torch.load("corpus.5fe0e8525c745b4a450533184c69e47e.data") #pennchar

In [5]:
loaded_corpus = torch.load("corpus.e4b3ed728d0b973ba43f49cb1e1d45f6.data") #enwik

In [ ]:
loaded_corpus

In [6]:
loaded_corpus.__dict__

{'dictionary': <data.Dictionary at 0x1c554312c50>,
 'train': tensor([ 0,  1,  2,  ..., 28, 26, 26]),
 'valid': tensor([12,  7,  5,  ..., 29, 26, 26]),
 'test': tensor([ 5,  7,  3,  ..., 16, 26, 26])}

In [4]:
len(loaded_corpus.dictionary)

50

In [6]:
len(loaded_corpus.dictionary)

205

In [8]:
loaded_corpus.dictionary.word2idx

{'a': 0,
 'e': 1,
 'r': 2,
 '_': 3,
 'b': 4,
 'n': 5,
 'k': 6,
 'o': 7,
 't': 8,
 'l': 9,
 'i': 10,
 'z': 11,
 'c': 12,
 'w': 13,
 'y': 14,
 'u': 15,
 's': 16,
 'f': 17,
 'm': 18,
 'g': 19,
 'h': 20,
 'd': 21,
 '-': 22,
 'q': 23,
 'p': 24,
 'x': 25,
 '<eos>': 26,
 '<': 27,
 '>': 28,
 'N': 29,
 'j': 30,
 'v': 31,
 '.': 32,
 "'": 33,
 '1': 34,
 '9': 35,
 '5': 36,
 '0': 37,
 '&': 38,
 '$': 39,
 '3': 40,
 '2': 41,
 '4': 42,
 '8': 43,
 '6': 44,
 '7': 45,
 '#': 46,
 '\\': 47,
 '/': 48,
 '*': 49}

In [6]:
loaded_corpus.dictionary.idx2word

['60',
 '109',
 '101',
 '100',
 '105',
 '97',
 '119',
 '107',
 '32',
 '120',
 '108',
 '110',
 '115',
 '61',
 '34',
 '104',
 '116',
 '112',
 '58',
 '47',
 '46',
 '111',
 '114',
 '103',
 '45',
 '48',
 '51',
 '50',
 '49',
 '88',
 '77',
 '76',
 '83',
 '99',
 '118',
 '62',
 '<eos>',
 '102',
 '87',
 '98',
 '95',
 '80',
 '54',
 '121',
 '84',
 '85',
 '52',
 '53',
 '73',
 '55',
 '56',
 '57',
 '72',
 '67',
 '65',
 '90',
 '117',
 '74',
 '35',
 '82',
 '69',
 '68',
 '91',
 '93',
 '123',
 '125',
 '70',
 '42',
 '39',
 '124',
 '106',
 '38',
 '113',
 '59',
 '122',
 '44',
 '71',
 '40',
 '41',
 '79',
 '75',
 '86',
 '66',
 '78',
 '195',
 '169',
 '63',
 '89',
 '170',
 '162',
 '160',
 '179',
 '81',
 '33',
 '226',
 '128',
 '148',
 '126',
 '147',
 '161',
 '37',
 '217',
 '132',
 '216',
 '167',
 '183',
 '136',
 '138',
 '208',
 '144',
 '189',
 '176',
 '209',
 '133',
 '184',
 '188',
 '175',
 '140',
 '178',
 '166',
 '219',
 '215',
 '168',
 '155',
 '153',
 '150',
 '157',
 '227',
 '130',
 '131',
 '173',
 '186',
 '23

In [9]:
loaded_corpus.train[:50]

tensor([ 0,  1,  2,  3,  4,  5,  6,  4,  7,  4,  8,  9,  1, 10, 11, 12, 13, 14,
        15, 16, 16, 17, 18, 19, 19,  6,  6,  6, 20,  1,  2,  3,  4,  5,  6,  4,
         7,  4, 20, 21, 22, 23, 19,  9,  1, 10, 19,  2,  9, 17])

In [7]:
loaded_corpus.train[:50]

tensor([ 0,  1,  2,  3,  4,  5,  6,  4,  7,  4,  8,  9,  1, 10, 11, 12, 13, 14,
        15, 16, 16, 17, 18, 19, 19,  6,  6,  6, 20,  1,  2,  3,  4,  5,  6,  4,
         7,  4, 20, 21, 22, 23, 19,  9,  1, 10, 19,  2,  9, 17])

In [9]:
"".join(list(map(lambda x:chr(int(loaded_corpus.dictionary.idx2word[x])),list(loaded_corpus.train[:100].numpy()))) )

'<mediawiki xmlns="http://www.mediawiki.org/xml/export-0.3/" xmlns:xsi="http://www.w3.org/2001/XMLSch'

In [9]:
len(loaded_corpus.train)

5017483

In [10]:
len(loaded_corpus.train)

90000001

In [6]: len(hidden[0]),len(hidden[1]),len(hidden[2])
Out[6]: (2, 2, 2)

In [7]: hidden[0][0].shape
Out[7]: torch.Size([1, 128, 1000])

In [8]: hidden[0][1].shape
Out[8]: torch.Size([1, 128, 1000])

In [9]: hidden[1][0].shape
Out[9]: torch.Size([1, 128, 1000])

In [10]: hidden[1][1].shape
Out[10]: torch.Size([1, 128, 1000])

In [11]: data.shape
Out[11]: torch.Size([148, 128])

In [12]: output.shape
Out[12]: torch.Size([18944, 200])

In [13]: 148*128
Out[13]: 18944

In [14]: hidden[2][0].shape
Out[14]: torch.Size([1, 128, 200])

In [15]: hidden[2][1].shape
Out[15]: torch.Size([1, 128, 200])

In [ ]:
# def tokenize_from_corpus(corp,path):
#     print("path {}".format(path))
#     assert os.path.exists(path)
#     # Add words to the dictionary
#     with open(path, 'r') as f:
#         tokens = 0
#         for line in f:
#             line_split = line.split(",")
#             parse_info = line_split[0].split(",")
#             info_dict = {}
#             for item in parse_info:
#                 item = item.strip()
#                 k,v = item.split("[")
#                 info_dict[k] = v[:-1]
#             if ""
#                 words = list() + ['<eos>']
#                 tokens += len(words)
#             #for word in words:
#             #    self.dictionary.add_word(word)

#     # Tokenize file content
#     with open(path, 'r') as f:
#         ids = torch.LongTensor(tokens)
#         token = 0
#         for line in f:
#             words = line.split() + ['<eos>']
#             for word in words:
#                 ids[token] = self.dictionary.word2idx[word]
#                 token += 1

#     return ids

In [ ]:
def tokenize_from_corpus(corp,path):
    print("path {}".format(path))
    assert os.path.exists(path)
    # Add words to the dictionary
    with open(path, 'r') as f:
        tokens = 0
        for line in f:
            line_split = line.split(",")
            parse_info = line_split[0].split(",")
            info_dict = {}
            for item in parse_info:
                item = item.strip()
                k,v = item.split("[")
                info_dict[k] = v[:-1]
            if ""
                words = list() + ['<eos>']
                tokens += len(words)
            #for word in words:
            #    self.dictionary.add_word(word)

    # Tokenize file content
    with open(path, 'r') as f:
        ids = torch.LongTensor(tokens)
        token = 0
        for line in f:
            words = line.split() + ['<eos>']
            for word in words:
                ids[token] = self.dictionary.word2idx[word]
                token += 1

    return ids

In [18]:
embedding = torch.nn.Embedding(10, 3)

In [19]:
inp = torch.LongTensor([[1,2,4,5],[4,3,2,9]])

In [20]:
embedding(inp)

tensor([[[ 1.6307,  0.5950, -1.0985],
         [ 0.4877, -0.5085,  0.4470],
         [ 0.4094, -0.0967, -0.6158],
         [-0.5101, -0.5067, -0.7284]],

        [[ 0.4094, -0.0967, -0.6158],
         [-1.4515,  1.6180,  2.0743],
         [ 0.4877, -0.5085,  0.4470],
         [-0.6140,  0.5353,  1.3370]]], grad_fn=<EmbeddingBackward0>)

In [21]:
embedding.weight

Parameter containing:
tensor([[ 2.1684, -1.2940,  0.7165],
        [ 1.6307,  0.5950, -1.0985],
        [ 0.4877, -0.5085,  0.4470],
        [-1.4515,  1.6180,  2.0743],
        [ 0.4094, -0.0967, -0.6158],
        [-0.5101, -0.5067, -0.7284],
        [ 0.4966, -1.4853, -2.2427],
        [ 0.1699, -0.5736, -1.0182],
        [-0.2071,  0.1879, -0.8973],
        [-0.6140,  0.5353,  1.3370]], requires_grad=True)

In [28]:
import os
os.listdir("data/e2e-dataset")

['README.md',
 'devset.csv',
 'testset.csv',
 'testset_w_refs.csv',
 'trainset.csv']

In [50]:
ord(",")

44

In [59]:
ord("\"")

34

In [60]:
print(chr(34))

"


In [14]:
from collections import Counter

In [25]:
info_dicts = []
with open("data/e2e-dataset/trainset.csv", 'r') as f:
    tokens = 0
    linenum=1
    for line in f:
        if linenum>1:
            info_dict = {}
            line_split = line.split(chr(34))
            parse_info = line_split[1].split(",")
            parse_info=parse_info[1:]
            #print(parse_info)
            nonchars = Counter()
            for c in line_split[1]:
                if c not in loaded_corpus.dictionary.word2idx:
                    nonchars[c] += 1
            if len(nonchars)>0:
                print(nonchars)
            info_dict = {}
            for item in parse_info:
                #print(item)
                item = item.strip()
                if len(item.split("["))==2:
                    k,v = item.split("[")
                    info_dict[k] = v[:-1]
                else:
                    print(line,item)
            info_dicts.append(info_dict)
        linenum += 1

Counter({' ': 12, '[': 5, ']': 5, ',': 4, 'T': 2, 'V': 1, 'R': 1, 'Â': 1, '£': 1, 'C': 1, 'Ã': 1, '©': 1, 'A': 1})
Counter({' ': 7, '[': 5, ']': 5, ',': 4, 'T': 2, 'C': 2, 'B': 2, 'E': 1, 'R': 1, 'Ã': 1, '©': 1})
Counter({' ': 13, '[': 8, ']': 8, ',': 7, 'T': 2, 'E': 1, 'J': 1, 'R': 1, 'Â': 1, '£': 1, 'F': 1, 'B': 1, 'K': 1})
Counter({' ': 8, '[': 6, ']': 6, ',': 5, 'T': 3, 'M': 1, 'F': 1, 'R': 1, 'Â': 1, '£': 1, 'S': 1})
Counter({' ': 8, '[': 5, ']': 5, ',': 4, 'F': 2, 'L': 1, 'T': 1, 'R': 1, 'B': 1})
Counter({' ': 6, '[': 5, ']': 5, ',': 4, 'H': 2, 'B': 1, 'E': 1, 'R': 1, 'C': 1})
Counter({' ': 7, '[': 5, ']': 5, ',': 4, 'F': 2, 'T': 1, 'R': 1, 'B': 1})
Counter({' ': 12, '[': 7, ']': 7, ',': 6, 'T': 2, 'R': 2, 'W': 1, 'J': 1, 'Â': 1, '£': 1, 'F': 1, 'I': 1, 'C': 1})
Counter({' ': 8, '[': 6, ']': 6, ',': 5, 'F': 2, 'A': 1, 'T': 1})
Counter({' ': 10, '[': 5, ']': 5, ',': 4, 'T': 1, 'P': 1, 'F': 1, 'R': 1})
Counter({' ': 7, '[': 5, ']': 5, ',': 4, 'F': 2, 'B': 1, 'C': 1, 'T': 1, 'S': 1}

Counter({' ': 13, '[': 6, ']': 6, ',': 5, 'T': 4, 'R': 2, 'F': 1, 'B': 1})
Counter({' ': 13, '[': 6, ']': 6, ',': 5, 'C': 2, 'F': 2, 'T': 1, 'G': 1, 'Ã': 1, '©': 1, 'R': 1})
Counter({'[': 3, ' ': 3, ']': 3, ',': 2, 'T': 1, 'P': 1, 'I': 1, 'R': 1})
Counter({' ': 8, '[': 6, ']': 6, ',': 5, 'F': 2, 'T': 1, 'W': 1, 'R': 1, 'Â': 1, '£': 1})
Counter({' ': 8, '[': 4, ']': 4, ',': 3, 'T': 1, 'P': 1, 'I': 1})
Counter({' ': 16, '[': 7, ']': 7, ',': 6, 'R': 2, 'E': 2, 'T': 1, 'B': 1, 'F': 1, 'H': 1, 'I': 1})
Counter({' ': 10, '[': 6, ']': 6, ',': 5, 'G': 1, 'M': 1, 'I': 1, 'R': 1, 'Â': 1, '£': 1, 'F': 1, 'A': 1, 'B': 1, 'O': 1})
Counter({' ': 11, '[': 6, ']': 6, ',': 5, 'C': 2, 'T': 1, 'I': 1, 'H': 1})
Counter({' ': 8, '[': 5, ']': 5, ',': 4, 'T': 1, 'W': 1, 'C': 1, 'R': 1, 'Â': 1, '£': 1, 'F': 1})
Counter({' ': 14, '[': 8, ']': 8, ',': 7, 'T': 2, 'F': 2, 'E': 1, 'R': 1, 'Â': 1, '£': 1, 'B': 1, 'K': 1})
Counter({' ': 5, '[': 4, ']': 4, ',': 3, 'C': 1, 'R': 1, 'A': 1, 'B': 1, 'O': 1})
Counter({' '

Counter({'[': 5, ']': 5, ',': 4, ' ': 4, 'G': 1, 'T': 1, 'I': 1, 'F': 1})
Counter({' ': 7, '[': 3, ']': 3, ',': 2, 'T': 1, 'E': 1, 'J': 1})
Counter({' ': 10, '[': 6, ']': 6, ',': 5, 'T': 2, 'G': 1, 'R': 1, 'Â': 1, '£': 1, 'F': 1, 'B': 1})
Counter({' ': 11, '[': 5, ']': 5, ',': 4, 'F': 2, 'R': 1, 'Â': 1, '£': 1, 'E': 1, 'H': 1, 'I': 1})
Counter({' ': 8, '[': 5, ']': 5, 'T': 4, ',': 4, 'D': 1, 'E': 1, 'F': 1, 'P': 1, 'A': 1})
Counter({' ': 8, '[': 4, ']': 4, ',': 3, 'C': 1, 'R': 1, 'Â': 1, '£': 1, 'A': 1, 'B': 1, 'O': 1})
Counter({' ': 9, '[': 5, ']': 5, ',': 4, 'T': 1, 'P': 1, 'E': 1, 'R': 1})
Counter({' ': 6, '[': 5, ']': 5, ',': 4, 'F': 3, 'L': 1, 'T': 1, 'R': 1})
Counter({' ': 11, '[': 6, ']': 6, ',': 5, 'G': 1, 'M': 1, 'E': 1, 'R': 1, 'Â': 1, '£': 1, 'F': 1, 'A': 1, 'B': 1, 'O': 1})
Counter({' ': 10, '[': 7, ']': 7, ',': 6, 'F': 2, 'T': 1, 'I': 1, 'R': 1, 'Â': 1, '£': 1})
Counter({' ': 11, '[': 7, ']': 7, ',': 6, 'F': 3, 'T': 1, 'R': 1})
Counter({' ': 16, '[': 7, ']': 7, ',': 6, 'R'

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



In [11]:
#filename = "data/e2e-dataset/trainset.csv"
#filename = "F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_train.txt"
def tokenize_e2e(loaded_corpus,filename):
    #add new chars to corpus
    with open(filename, 'r') as f:
        tokens = 0
        for line in f:
            words = list(line) + ['<eos>']
            tokens += len(words)
            for word in words:
                loaded_corpus.dictionary.add_word(word)

    with open(filename, 'r') as f:
        ids = torch.LongTensor(tokens)
        token = 0
        for line in f:
            words = list(line) + ['<eos>']
            for word in words:
                ids[token] = loaded_corpus.dictionary.word2idx[word]
                token += 1

    return ids

def tokenize_e2e_enwik(loaded_corpus,filename):
    #add new chars to corpus
    with open(filename, 'r') as f:
        tokens = 0
        for line in f:
            words = list(line) + ['<eos>']
            tokens += len(words)
            for word in words:
                #print("word: ",word)
                word_as_int = str(ord(word)) if word!='<eos>' else word
                loaded_corpus.dictionary.add_word(word_as_int)

    with open(filename, 'r') as f:
        ids = torch.LongTensor(tokens)
        token = 0
        for line in f:
            words = list(line) + ['<eos>']
            for word in words:
                word_as_int = str(ord(word)) if word!='<eos>' else word
                ids[token] = loaded_corpus.dictionary.word2idx[word_as_int]
                token += 1

    return ids

In [36]:
loaded_corpus.dictionary.counter

Counter({0: 375300,
         1: 532241,
         2: 296657,
         3: 987381,
         4: 69607,
         5: 375061,
         6: 87427,
         7: 335805,
         8: 404580,
         9: 185267,
         10: 328750,
         11: 2822,
         12: 154270,
         13: 67909,
         14: 75308,
         15: 173745,
         16: 317566,
         17: 100016,
         18: 119219,
         19: 85204,
         20: 190729,
         21: 167741,
         22: 6969,
         23: 6024,
         24: 98237,
         25: 11939,
         26: 49202,
         27: 53299,
         28: 53299,
         29: 37607,
         30: 9146,
         31: 44733,
         32: 19494,
         33: 17414,
         34: 440,
         35: 317,
         36: 89,
         37: 622,
         38: 1616,
         39: 8982,
         40: 220,
         41: 112,
         42: 46,
         43: 103,
         44: 51,
         45: 61,
         46: 173,
         47: 75,
         48: 49,
         49: 26})

In [37]:
loaded_corpus.dictionary.total

5852950

In [23]:
loaded_corpus.dictionary.total

100000003

In [15]:
loaded_corpus.dictionary.counter = Counter()
loaded_corpus.dictionary.total = 0

In [16]:
#loaded_corpus.train = tokenize_e2e(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_train.txt")
loaded_corpus.train = tokenize_e2e_enwik(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_train.txt")

In [40]:
loaded_corpus.dictionary.total

9393725

In [17]:
#loaded_corpus.valid = tokenize_e2e(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_validation.txt")
loaded_corpus.valid = tokenize_e2e_enwik(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_validation.txt")

In [42]:
loaded_corpus.dictionary.total

10560768

In [18]:
#loaded_corpus.test = tokenize_e2e(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_test.txt")
loaded_corpus.test = tokenize_e2e_enwik(loaded_corpus,"F:/DellGL3/School/CMU/classes/Fall2022/10708/Project/gpt2/e2e_nlg/data/e2e_nlg_test.txt")

In [44]:
loaded_corpus.dictionary.total

11876272

In [54]:
loaded_corpus.train

tensor([ 5,  0, 18,  ..., 32, 66, 26])

In [39]:
loaded_corpus.train

tensor([ 11,   5,   1,  ...,  20, 205,  36])

In [19]:
"".join(list(map(lambda x:chr(int(loaded_corpus.dictionary.idx2word[x])),list(loaded_corpus.train[:100].numpy()))))

'name[The Vaults], eatType[pub], priceRange[more than £30], customer rating[5 out of 5], near[Café '

In [45]:
loaded_corpus.dictionary.word2idx

{'a': 0,
 'e': 1,
 'r': 2,
 '_': 3,
 'b': 4,
 'n': 5,
 'k': 6,
 'o': 7,
 't': 8,
 'l': 9,
 'i': 10,
 'z': 11,
 'c': 12,
 'w': 13,
 'y': 14,
 'u': 15,
 's': 16,
 'f': 17,
 'm': 18,
 'g': 19,
 'h': 20,
 'd': 21,
 '-': 22,
 'q': 23,
 'p': 24,
 'x': 25,
 '<eos>': 26,
 '<': 27,
 '>': 28,
 'N': 29,
 'j': 30,
 'v': 31,
 '.': 32,
 "'": 33,
 '1': 34,
 '9': 35,
 '5': 36,
 '0': 37,
 '&': 38,
 '$': 39,
 '3': 40,
 '2': 41,
 '4': 42,
 '8': 43,
 '6': 44,
 '7': 45,
 '#': 46,
 '\\': 47,
 '/': 48,
 '*': 49,
 '[': 50,
 'T': 51,
 ' ': 52,
 'V': 53,
 ']': 54,
 ',': 55,
 'R': 56,
 'Â': 57,
 '£': 58,
 'C': 59,
 'Ã': 60,
 '©': 61,
 'A': 62,
 'G': 63,
 'E': 64,
 'P': 65,
 '\n': 66,
 'B': 67,
 'D': 68,
 'J': 69,
 'F': 70,
 'K': 71,
 'M': 72,
 'S': 73,
 'L': 74,
 'H': 75,
 'Y': 76,
 'W': 77,
 'I': 78,
 'O': 79,
 'Z': 80,
 'U': 81,
 ';': 82,
 ':': 83,
 '?': 84,
 'Q': 85,
 '!': 86,
 '+': 87,
 '"': 88}

In [20]:
import hashlib

In [49]:
'corpus.{}.data'.format(hashlib.md5("data/enwik8".encode()).hexdigest())

'corpus.e4b3ed728d0b973ba43f49cb1e1d45f6.data'

In [53]:
e2e_corpus_path = 'corpus.{}.data'.format(hashlib.md5("e2e_path".encode()).hexdigest())
e2e_corpus_path

'corpus.92ba3b603ad40cc094c17f41be3f1354.data'

In [21]:
e2e_corpus_path = 'corpus.{}.data'.format(hashlib.md5("e2e_path_enwik".encode()).hexdigest())
e2e_corpus_path

'corpus.78a475e0154e6bad4f1db2bc8683d638.data'

In [22]:
torch.save(loaded_corpus,e2e_corpus_path)

In [18]:
line_split[1]

'name[Aromi], eatType[restaurant], food[English], area[city centre]'

In [19]:
loaded_corpus.dictionary.word2idx["["]

KeyError: '['

In [63]:
info_dicts[10]

{'food': 'Fast food',
 'area': 'riverside',
 'familyFriendly': 'no',
 'near': 'The Sorrento'}

In [46]:
list(info_dicts[0].keys())[0][0]

'"'

In [29]:
"a" in loaded_corpus.dictionary.word2idx

True

In [26]:
#loaded_corpus.dictionary.add_word("<GEN>")

50

In [27]:
#loaded_corpus.dictionary.add_word("<UNK>")

51